webpack/components/ProxmoxContainer/components/NetworkInterface.js

```diff
@@ -0,0 +1,141 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { Title, Divider, PageSection } from '@patternfly/react-core';
+import InputField from '../../common/FormInputs';
+import ProxmoxComputeSelectors from '../../ProxmoxComputeSelectors';
+import { translate as __ } from 'foremanReact/common/I18n';
+
+const NetworkInterface = ({
+  id,
+  networks,
+  bridges,
+  data,
+  updateNetworkData,
+}) => {
+  const [network, setNetwork] = useState(data);
+
+  useEffect(() => {
+    const currentNetData = JSON.stringify(network);
+    const parentNetData = JSON.stringify(data);
+
+    if (currentNetData !== parentNetData) {
+      updateNetworkData(id, network);
+    }
+  }, [network, id, data, updateNetworkData]);
+  const handleChange = e => {
+    const { name, type, checked } = e.target;
+    const value = type === 'checkbox' ? (checked ? '1' : '0') : e.target.value;
+    const updatedKey = Object.keys(network).find(
+      key => network[key].name === name
+    );
+    const updatedData = {
+      ...network,
+      [updatedKey]: { ...network[updatedKey], value },
+    };
+    setNetwork(updatedData);
+  };
+  const bridgesMap = bridges.map(bridge => ({
+    value: bridge.iface,
+    label: bridge.iface,
+  }));
+
+  return (
+    <div style={{ position: 'relative' }}>
+      <Divider component="li" style={{ marginBottom: '2rem' }} />
+      <InputField
+        name={network.id.name}
+        label={__('Indentifier')}
+        info={__('net[n] with n integer >= 0, e.g. net0')}
+        type="text"
+        value={network.id.value}
+        onChange={handleChange}
+      />
+      <InputField
+        name={network.name.name}
+        label={__('Name')}
+        info={__('eth[n] with n integer >= 0, e.g. eth0')}
+        type="text"
+        value={network.name.value}
+        onChange={handleChange}
+      />
+      <InputField
+        name={network.bridge.name}
+        label={__('Bridge')}
+        type="select"
+        options={bridgesMap}
+        value={network.bridge.value}
+        onChange={handleChange}
+      />
+      <InputField
+        name={network?.dhcp?.name}
+        label={__('DHCP IPv4')}
+        type="checkbox"
+        value={network?.dhcp?.value}
+        checked={network?.dhcp?.value === '1'}
+        onChange={handleChange}
+      />
+      <InputField
+        name={network?.cidr?.name}
```

Comment:
Why do you have the optional chaining operator in some of the InputFields and in others you don't?

Reply:
I added the chaining operator to all now to have same rendering in case network/cidr is nil.